cluster/examples/kubernetes/ceph/cluster-on-pvc.yaml

```diff
@@ -145,6 +146,14 @@ spec:
       #       - ReadWriteOnce
       # Scheduler name for OSD pod placement
       # schedulerName: osd-scheduler
+  resources:
```

Comment:
It's not clear to me why we have 2 `resources` sections. Can't we merge both with something like:

```yaml
  #  osd:
  #    limits:
  #      cpu: "200m"
  #      memory: "200Mi"
  #   requests:
  #      cpu: "200m"
  #      memory: "200Mi"
  #
  #  prepareosd:
  #    limits:
  #      cpu: "200m"
  #      memory: "200Mi"
  #   requests:
  #      cpu: "200m"
  #      memory: "200Mi"
```

Reply:
Agreed that it is awkward. The reason is that the resources are applied for each `storageClassDeviceSet`, which could have different values. I really don't know why someone would do that, so we should likely deprecate the resources there and just use the top-level resources. Let's do that with a separate PR though.

Reply:
Sounds good as a follow-up!

Reply:
hmm....
A deviceSet could be on different hardware. That includes number of drives per node. So the configuration of an osd's dedicated memory/cpu may be significantly different on one node to another. While I don't yet have a configuration that would benefit from it, I could see how it would be important to support for when new hardware comes in and needs to join the cluster.

Reply:
@leseb and I discussed further as well and agreed there may be scenarios where different resources may be required if device sets have different perf characteristics, so we'll leave it this way.